### PR TITLE
Add function and documentation to draw circle

### LINKED ISSUE #8083: Missing ImageDraw.Draw.circle function
### What did you do?

I find it frustrating that Pillow's ImageDraw does still not have a function to draw a circle. Every time I come around to do it, I find that I must use ImageDraw.Draw.ellipse instead, which costs time. It must be even more frustrating for beginners who don't know more complex shapes like ellipses.

### What did you expect to happen?

An ImageDraw.Draw.circle to be available

### What actually happened?

It does not exist

### What are your OS, Python and Pillow versions?

* OS: Every
* Python: Every
* Pillow: Every

### PR DESCRIPTION
**Pull request by void4**
Resolves #8083

Implements the ImageDraw.circle feature as described and desired in the issue

### CODE REVIEW

File: src/PIL/ImageDraw.py [hunk 1]
```diff
@@ -181,6 +181,15 @@ def ellipse(self, xy: Coords, fill=None, outline=None, width=1) -> None:
...
             self.draw.draw_ellipse(xy, ink, 0, width)
 
+    def circle(self, xy: Coords, radius, fill=None, outline=None, width=1) -> None:
+        """Draw a circle given center coordinates and a radius."""
+        ink, fill = self._getink(outline, fill)
+        ellipse_xy = (xy[0] - radius, xy[1] - radius, xy[0] + radius, xy[1] + radius)
+        if fill is not None:
+            self.draw.draw_ellipse(ellipse_xy, fill, 1)
+        if ink is not None and ink != fill and width != 0:
+            self.draw.draw_ellipse(ellipse_xy, ink, 0, width)
```
Comment (Yay295) [inline src/PIL/ImageDraw.py:191]: These four lines could be replaced by just `self.ellipse(ellipse_xy, fill, outline, width)`, and you can then remove the first line of code as well.

File: docs/reference/ImageDraw.rst [hunk 2]
```diff
@@ -240,6 +240,19 @@ Methods
...
+    Draws a circle given the center coordinates and a radius.
+
+    :param xy: One point to define the circle center. Sequence:
+        ``[x, y]``
+    :param radius: Radius of the circle
+    :param outline: Color to use for the outline.
+    :param fill: Color to use for the fill.
+    :param width: The line width, in pixels.
+
+        .. versionadded:: ?.?.?
```
Comment (hugovk) [inline docs/reference/ImageDraw.rst:254]:
```suggestion
        .. versionadded:: 10.4.0
```

**Review by hugovk — CHANGES_REQUESTED** [review]
Please could you add unit tests, and mention this in the release notes?

File: docs/reference/ImageDraw.rst [hunk 3]
```diff
@@ -240,6 +240,19 @@ Methods
 
         .. versionadded:: 5.3.0
 
+.. py:method:: ImageDraw.circle(xy, radius, fill=None, outline=None, width=1)
+
+    Draws a circle given the center coordinates and a radius.
```
Comment (hugovk) [inline docs/reference/ImageDraw.rst:245]: Let's move this up to indicate the function is new:
```suggestion
    Draws a circle given the center coordinates and a radius.
    
    .. versionadded:: 10.4.0
```

File: docs/reference/ImageDraw.rst [hunk 4]
```diff
@@ -240,6 +240,19 @@ Methods
...
+
+    :param xy: One point to define the circle center. Sequence:
+        ``[x, y]``
+    :param radius: Radius of the circle
+    :param outline: Color to use for the outline.
+    :param fill: Color to use for the fill.
+    :param width: The line width, in pixels.
+
+        .. versionadded:: 10.4.0
+
```
Comment (hugovk) [inline docs/reference/ImageDraw.rst:255]: And not the `width` param:
```suggestion
```

File: src/PIL/ImageDraw.py [hunk 5]
```diff
@@ -181,6 +181,11 @@ def ellipse(self, xy: Coords, fill=None, outline=None, width=1) -> None:
         if ink is not None and ink != fill and width != 0:
             self.draw.draw_ellipse(xy, ink, 0, width)
 
+    def circle(self, xy: Coords, radius, fill=None, outline=None, width=1) -> None:
```
Comment (radarhere) [inline src/PIL/ImageDraw.py:184]:
```suggestion
    def circle(
        self, xy: Sequence[float], radius: float, fill=None, outline=None, width=1
    ) -> None:
```
`Coords` is either a sequence or a sequence of a sequence. Here, you just want a sequence.
https://github.com/python-pillow/Pillow/blob/bbe1effd63149eb3d43d59efb5fe2659f93700d0/src/PIL/_typing.py#L20

**Comment by void4** [comment]
> Please could you add unit tests, and mention this in the release notes?

I've added it to the release notes but am not quite sure how the unit tests work, especially the parametrization :/

**Comment by void4** [comment]
Thank you - merged :)

**Review by radarhere — APPROVED** [review]

**Comment by hugovk** [comment]
Thank you!

**Pull request merged**